web/source/kmwcallback.ts

```diff
@@ -1108,44 +1108,52 @@ namespace com.keyman {
               if(this.keymanweb.util.activeDevice.touchable != (constraint == 'touch')) {
                 result=false;
               }
-          }
+              break;
 
-          switch(constraint) {
+            case 'macos':
+            case 'mac':
+              constraint = 'macosx';
+              // fall through
+            case 'macosx':
             case 'windows':
             case 'android':
             case 'ios':
-            case 'macosx':
             case 'linux':
               if(this.keymanweb.util.activeDevice.OS.toLowerCase() != constraint) {
                 result=false;
               }
-          }
-
-          switch(constraint) {
+              break;
+          
             case 'tablet':
             case 'phone':
             case 'desktop':
-              if(this.keymanweb.util.activeDevice.formFactor != constraint) {
+              if(this.keymanweb.util.device.formFactor != constraint) {
                 result=false;
               }
-          }
+              break;
 
-          switch(constraint) {
             case 'web':
-              if(this.keymanweb.util.activeDevice.browser == 'native') {
+              if(this.keymanweb.util.device.browser == 'native') {
                 result=false; // web matches anything other than 'native'
               }
               break;
+              
             case 'native':
+              // This will return true for embedded KeymanWeb
             case 'ie':
             case 'chrome':
             case 'firefox':
             case 'safari':
             case 'opera':
```

Comment:
is this missing `edge`? 
When I run the local testing/platform page on Edge 42.17134.1.0

`y` is saying `[Browser Undefined]`

Reply:
Good catch

Reply:
I added `edge` locally, but my `y` still reports `[Browser Undefined]` :(